README.md

```diff
@@ -54,6 +54,165 @@ public class GetSampleEndpoint(ISampleService sampleService) : RadEndpoint<GetSa
+- Constructor dependency injection (scoped lifetime)
+- Built-in conveniences: `HttpContext`, `Logger`, `Env`, `Response`
+- TypedResult shortcuts: `Send()`, `SendNotFound()`, `SendProblem()`, etc.
+- Automatic validation via FluentValidation
+
+#### Response Methods
+RadEndpoints provides strongly-typed response methods that return ASP.NET Core TypedResults:
+
+**Success Responses**
+```csharp
+// 200 OK with response body
+Response = new UserResponse { Id = 1, Name = "John" };
+Send(); // or Send(customResponse)
+
+// 201 Created with Location header
+SendCreatedAt("/users/123", response);
+```
+
+**Client Error Responses**
+```csharp
+// 400 Bad Request - Validation errors
+SendValidationError("Field is required");
+
+// 404 Not Found
+SendNotFound("User not found");
+
+// 409 Conflict
+SendConflict("Email already exists");
+```
+
+**Server Error Responses**
+```csharp
+// 500 Internal Server Error
+SendInternalError("Database connection failed");
+
+// 502 Bad Gateway
+SendExternalError("External API unreachable");
+
+// 504 Gateway Timeout
+SendExternalTimeout("External API timeout");
+```
+
+**Redirect Responses**
+```csharp
+// 301 Moved Permanently
+SendRedirect("/new-location", permanent: true);
+
+// 302 Found (temporary redirect)
+SendRedirect("/temporary-location");
+
+// 307/308 with method preservation
+SendRedirect("/location", permanent: true, preserveMethod: true);
+```
+
+**Binary Responses**
+```csharp
+// Byte array download
+SendBytes(new RadBytes 
+{ 
+    Bytes = data, 
+    ContentType = "application/pdf",
+    FileDownloadName = "report.pdf"
+});
+
+// Stream response
+SendStream(new RadStream 
+{ 
+    Stream = fileStream, 
+    ContentType = "text/csv"
+});
+
+// Physical file
+SendFile(new RadFile 
+{ 
+    Path = "/path/to/file.txt",
+    ContentType = "text/plain"
+});
+```
+
+**Custom Problem Details**
+```csharp
+// Using ProblemHttpResult
+SendProblem(TypedResults.Problem(
+    title: "Custom error",
+    statusCode: 422,
+    detail: "Detailed error message"));
+
+// Using IRadProblem for domain errors
+SendProblem(new CustomDomainError("Business rule violated"));
+```
+
+**Testing Response Methods**
+
+Integration tests use typed client extensions:
+```csharp
+// Test string responses (NotFound, Conflict)
+var response = await client.GetAsync<NotFoundEndpoint, Request, string>(request);
+response.Http.StatusCode.Should().Be(HttpStatusCode.NotFound);
+
+// Test ProblemDetails responses
+var response = await client.GetAsync<ErrorEndpoint, Request, ProblemDetails>(request);
+response.Content.Title.Should().Be("Internal server error");
+
+// Test binary responses (use raw HttpClient)
+var response = await client.GetAsync("/api/download");
+var bytes = await response.Content.ReadAsByteArrayAsync();
+```
+
+See the test endpoints in [`/MinimalApi/Features/ResultEndpoints`](https://github.com/MetalHexx/RadEndpoints/tree/main/MinimalApi/Features/ResultEndpoints) for complete examples of all response methods.
+
```

Comment:
You didn't mention usage of the Untyped RadEndpoint class that can be used a little bit more like normal MinimalEndpoints.  Explain the features you still get using the basic base class.  There good intofrmation in the TypedResultsPutEndpoint.  Also mention the ability to create Custom base endpoints, see: CustomGetEndpoint and CustomBaseEndpoint